add-chain/main.go

```diff
@@ -230,3 +261,32 @@ func getGasPayingToken(l1rpcURl string, SystemConfigAddress superchain.Address)
 
 	return (*superchain.Address)(&result.Addr), nil
 }
+
+func copyDeployConfigFile(sourcePath string, targetDir string) error {
+	data, err := os.ReadFile(sourcePath)
+	if err != nil {
+		return err
+	}
+	return os.WriteFile(path.Join(targetDir, "deploy-config.json"), data, os.ModePerm)
+}
+
+func writeGenesisValidationMetadata(commit string, targetDir string) error {
+	// Define default metadata params:
+	// These may not be sufficient to make the genesis validation work,
+	// but we address that with some manual trial-and-error intervention
+	// involving OPLabs engineers after the add-chain command runs.
+	const defaultNodeVersion = "18.12.1"
+	const defaultMonorepoBuildCommand = "pnpm"
```

Comment:
Similar to the monorepo commit hash, could these two fields be derived and written into one of the pre-existing inputs to the `add-chain` tooling (i.e. the deploy-config or rollup.json file)? That way we are much more likely to be both backwards and forwards compatible, without requiring manual intervention

Reply:
In principle yes, although I think we are more likely to explore more radical overhauls of the genesis creation path which will make this unnecessary. I'd be a bit nervous about modifying the existing deploy-config.json file, as it is one of the inputs to the genesis creation tool and it seems an antipattern to modify the input files.